great_expectations/compatibility/pyspark.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+from great_expectations.compatibility.not_imported import NotImported
+
+SPARK_NOT_IMPORTED = NotImported(
+    "pyspark is not installed, please 'pip install pyspark'"
+)
+
+try:
+    import pyspark  # noqa: TID251
```

Comment:
You shouldn't need the # noqa: TID251 for any of the compatibility files. That is what the `[tool.ruff.per-file-ignores]` directives are for. In other words, by adding the per file ignore for TID251 we are allowing import in the compatibility files and disallowing it elsewhere (to make sure we use the compatibility version).

Reply:
(This is applicable to the other files as well)

Reply:
@anthonyburdi I agree, but it did not work.  I can try again.

Reply:
@anthonyburdi Amazing -- I tried removing `# noqa: TID251` "escapes" and it seems to be working fine now.  I wonder if also deleting the `AttributeError` from the applicable `except` clauses had something to do with it.  Thanks.